runtime/interpreter/value.go

```diff
@@ -16533,27 +16652,40 @@ func (*EphemeralReferenceValue) DeepRemove(_ *Interpreter) {
 //
 type AddressValue common.Address
 
-func NewAddressValue(a common.Address) AddressValue {
-	return NewAddressValueFromBytes(a[:])
-}
-
-func NewAddressValueFromBytes(b []byte) AddressValue {
+func NewUnmeteredAddressValue(b []byte) AddressValue {
 	result := AddressValue{}
 	copy(result[common.AddressLength-len(b):], b)
 	return result
 }
 
+func NewAddressValue(
+	memoryGauge common.MemoryGauge,
+	address common.Address,
+) AddressValue {
+	return NewAddressValueFromBytes(
+		memoryGauge,
+		address[:],
```

Comment:
Same here, go internally allocates a new slice for `address[:]`. Need to pass this as an anon-func

Reply:
I think it's fine.

This call should only happen when the address is at most 8 bytes long so we can treat it as a constant. Constants use a known, smallish amount of memory, so it's safe to allocate them ahead of time.

Reply:
Updated to allocate first.

Reply:
Actually, the suggestion here was to do the metering first and then the allocation. That is what we agreed earlier: https://github.com/dapperlabs/cadence-private-issues/issues/2#issuecomment-1065622097